modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -241,6 +241,156 @@ PERF_TEST_P_(BackgroundSubtractorPerfTest, TestPerformance)
 
 //------------------------------------------------------------------------------
 
+inline void generateInputKalman(const int mDim, const MatType2& type,
+                                const size_t testNumMeasurements, const bool receiveRandMeas,
+                                std::vector<bool>&    haveMeasurements,
+                                std::vector<cv::Mat>& measurements)
+{
+    cv::RNG& rng = cv::theRNG();
+    measurements = std::vector<cv::Mat>(testNumMeasurements, cv::Mat::zeros(mDim, 1, type));
+    haveMeasurements = std::vector<bool>(testNumMeasurements, true);
+    cv::Mat measurement(mDim, 1, type);
+    for (std::size_t i = 0; i < testNumMeasurements; i++)
+    {
+        if (receiveRandMeas)
+        {
+            haveMeasurements[i] = rng(2u) == 1; // returns 0 or 1 - whether we have measurement
+                                                // at this iteration or not
+        } // if not - testing the slowest case in which we have measurements at every iteration
+
+        if (haveMeasurements[i])
+        {
+            cv::randu(measurement, cv::Scalar::all(-1), cv::Scalar::all(1));
```

Comment:
Why don't you use measurements[i] here?

Reply:
Right! moreover, it wasn't correct! I was initializing the vector fully with zeros, and then pushing back to it.
Fixed